internal/controllers/subnet/actuator.go

```diff
@@ -272,12 +275,190 @@ func (actuator subnetActuator) DeleteResource(ctx context.Context, obj orcObject
+
+		found := false
+		for _, pool := range osResource.AllocationPools {
+			if pool.Start == allocationPools[i].Start &&
+				pool.End == allocationPools[i].End {
+				found = true
+				break
+			}
+		}
+		if !found {
+			missingAllocationPool = true
+		}
+	}
+
+	extraAllocationPool := false
+	for i := range osResource.AllocationPools {
+		found := false
+		for _, pool := range allocationPools {
+			if pool.Start == osResource.AllocationPools[i].Start &&
+				pool.End == osResource.AllocationPools[i].End {
+				found = true
+				break
+			}
+		}
+		if !found {
+			extraAllocationPool = true
+		}
+	}
+
+	// If the spec doesn't set allocation pools, we'll get a default one and should not try to update it.
+	if len(resource.AllocationPools) > 0 && (missingAllocationPool || extraAllocationPool) {
+		updateOpts.AllocationPools = allocationPools
+	}
+}
+
+func handleHostRoutesUpdate(updateOpts *subnets.UpdateOpts, resource *resourceSpecT, osResource *osResourceT) {
+	missingHostRoute := false
+	hostRoutes := make([]subnets.HostRoute, len(resource.HostRoutes))
+	for i := range resource.HostRoutes {
+		hostRoutes[i].DestinationCIDR = string(resource.HostRoutes[i].Destination)
+		hostRoutes[i].NextHop = string(resource.HostRoutes[i].NextHop)
+
+		found := false
+		for _, route := range osResource.HostRoutes {
+			if route.DestinationCIDR == hostRoutes[i].DestinationCIDR &&
+				route.NextHop == hostRoutes[i].NextHop {
+				found = true
+				break
+			}
+		}
+		if !found {
+			missingHostRoute = true
+		}
+	}
+
+	extraHostRoute := false
+	for i := range osResource.HostRoutes {
+		found := false
+		for _, route := range hostRoutes {
+			if route.DestinationCIDR == osResource.HostRoutes[i].DestinationCIDR &&
+				route.NextHop == osResource.HostRoutes[i].NextHop {
+				found = true
+				break
+			}
+		}
+		if !found {
+			extraHostRoute = true
+		}
+	}
+
+	if missingHostRoute || extraHostRoute {
+		updateOpts.HostRoutes = &hostRoutes
+	}
+}
+
+func handleDNSNameserversUpdate(updateOpts *subnets.UpdateOpts, resource *resourceSpecT, osResource *osResourceT) {
+	nameservers := make([]string, len(resource.DNSNameservers))
+	for i := range resource.DNSNameservers {
+		nameservers[i] = string(resource.DNSNameservers[i])
+	}
+
+	// Let's not bother about potential duplicate entries: they will be rejected by neutron API
+	if !slices.Equal(osResource.DNSNameservers, nameservers) {
+		updateOpts.DNSNameservers = &nameservers
+	}
+}
+
+func handleEnableDHCPUpdate(updateOpts *subnets.UpdateOpts, resource *resourceSpecT, osResource *osResourceT) {
+	// Default is true
+	enableDHCP := ptr.Deref(resource.EnableDHCP, true)
+	if osResource.EnableDHCP != enableDHCP {
+		updateOpts.EnableDHCP = &enableDHCP
+	}
+}
+
+func handleGatewayUpdate(updateOpts *subnets.UpdateOpts, resource *resourceSpecT, osResource *osResourceT) {
+	if resource.Gateway != nil {
+		switch resource.Gateway.Type {
+		case orcv1alpha1.SubnetGatewayTypeAutomatic:
+			// Nothing to do
```

Comment:
What if I change from `SubnetGatewayTypeIP` to `SubnetGatewayTypeAutomatic`?

Reply:
Nothing. Neutron notices it already has an gateway and will not change it.